src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriBuilder.java

```diff
@@ -782,14 +787,72 @@ public boolean addSourceInfo(
 
             // Ok we have channel for this content, let's add sources
             ColibriConferenceIQ.Channel requestChannel
-                = getRequestChannel(contentName, channel.getID());
+                = getRequestChannel(contentName, channel);
 
             addSources(requestChannel, sourceMap.get(contentName));
         }
 
         return hasAnyChanges;
     }
 
+    /**
+     * Sets the Octo relays for specific channels in the request currently being
+     * built.
+     *
+     * @param octoRelays the map of content name to the list of
+     * <tt>SourcePacketExtension</tt>.
+     * @param localChannelsInfo {@link ColibriConferenceIQ} holding info about
+     * Colibri channels to be updated.
+     *
+     * @return <tt>true</tt> if the request yields any changes in Colibri
+     * channels state on the bridge or <tt>false</tt> otherwise. In general when
+     * <tt>false</tt> is returned for all combined requests it makes no sense
+     * to send it.
+     */
+    public boolean addOctoRelays(
+        List<String> octoRelays,
+        ColibriConferenceIQ localChannelsInfo)
+    {
+        Objects.requireNonNull(octoRelays, "octoRelays");
+        Objects.requireNonNull(localChannelsInfo, "localChannelsInfo");
+
+        if (conferenceState == null
+            || StringUtils.isNullOrEmpty(conferenceState.getID()))
+        {
+            // We are not initialized yet
+            return false;
+        }
+
+        assertRequestType(RequestType.CHANNEL_INFO_UPDATE);
```

Comment:
i know this isn't new code, but i don't get what this is doing?  what is it asserting on/against?

Reply:
There are two main types of requests: requests for new channels (ALLOCATE_REQUEST) and updating state for existing channels (CHANNEL_INFO_UPDATE). The API doesn't allow to mix the two, e.g. if we are building an allocate request, we don't know the IDs of the channels yet, so we can not use the addOctoRelays method (since it wouldn't know how to identify the channels to add relays to).

Reply:
ok, i looked up the method and see it compares it against a member variable, and actually even re-assigns that member variable in `assertRequestType` 😮 .  not a big fan of the naming/semantics of that but i know it was already there so don't worry about it for this.